cephadm: support for CA signed keys (#51899)
ceph/ceph: doc/cephadm/install.rst

      cephadm --image *<hostname>*:5000/ceph/ceph bootstrap --mon-ip *<mon-ip>*

.. _cluster network: ../rados/configuration/network-config-ref#cluster-network

.. _cephadm-bootstrap-custom-ssh-keys:

Deployment with custom SSH keys
-------------------------------

> Bootstrap allows users to create their own private/public SSHkey pair

Reviewer: ```suggestion
Bootstrap allows users to create their own private/public SSH key pair
```

> rather than having cephadm generate them itself

Reviewer: ```suggestion
rather than having cephadm generate them itself.
```

Maybe even
```suggestion
rather than having cephadm generate them automatically.
```

To use custom SSH keys, pass the ``--ssh-private-key`` and ``--ssh-public-key``

> fields to bootstrap. The parameters expect a filepath to the file where the

Reviewer: ```suggestion
fields to bootstrap. Both parameters require a path to the file where the
```

keys are stored. For example:

.. prompt:: bash #

>   cephadm bootstrap --mon-ip <ip-addr> --ssh-private-key <private-key-filepath> --ssh-public-key <public-key-filepath>

Reviewer: This is less an example than a template but it's not a biggie IMO.

   to the cluster, make sure the public key of the key pair provided is setup
   as an authorized key for the ssh user being used, typically root. If you'd
   like more info on using a non-root user as the ssh user, see :ref:`cephadm-bootstrap-further-info`

.. _cephadm-bootstrap-ca-signed-keys:

Deployment with CA signed SSH keys
----------------------------------

As an alternative to standard public key authentication, cephadm also supports

> deployment using CA signed keys. Before bootstrapping it's recommended to setup

Reviewer: ```suggestion
deployment using CA signed keys. Before bootstrapping it's recommended to set up
```

> the CA public key as a trusted CA key on hosts you'd like to add to eventually

Author: yes, what I wrote is definitely not grammatically correct here

add to the cluster. For example:

.. prompt:: bash

>   # going to be our own CA here, meaning we'll need to make a CA key

Author: Definitely was missing at lest a "We're" at the beginning of the sentence or something. I'll go with your suggestion.
